stdlib/REPL/test/repl.jl

```diff
@@ -16,8 +16,26 @@ include(joinpath(BASE_TEST_PATH, "testenv.jl"))
 include("FakeTerminals.jl")
 import .FakeTerminals.FakeTerminal
 
+
+function kill_timer(delay)
+    # Give ourselves a generous timer here, just to prevent
+    # this causing e.g. a CI hang when there's something unexpected in the output.
+    # This is really messy and leaves the process in an undefined state.
+    # the proper and correct way to do this in real code would be to destroy the
+    # IO handles: `close(stdout_read); close(stdin_write)`
+    test_task = current_task()
+    function kill_test()
+        # **DON'T COPY ME.**
+        # The correct way to handle timeouts is to close the handle:
+        # e.g. `close(stdout_read); close(stdin_write)`
+        schedule(test_task, "hard kill repl test"; error=true)
+        print(stderr, "WARNING: attempting hard kill of repl test after exceeding timeout\n")
+    end
+    return Timer(kill_test, delay)
+end
```

Comment:
Instead of this `kill_timer`, I think it might make more sense for us to make a `readuntil_timed(io, needle; timeout=10)` or something.  That will allow us to create a closure within `readuntil` that would give us a useful backtrace, and would apply to all the `readuntil()` calls within this file, whereas this `kill_test()` doesn't apply to all of them (just most of them).

Reply:
This `kill_test` very aggressively applies to everything in this file (in a racy, panic-then-destroy-the-world-but-with-a-backtrace sort of way). As the comments above repeatedly say, the proper way to do this, in a place where you care about possible error recovery, is to close the `io` object.

Although maybe I should make the timeout much longer here. I've been doing some more testing, and it realized it could get pretty annoying to have your debugging session get killed if it lasts more than a couple minutes (as just happened to me). But then again, that's really just the price you have to pay for ever considering putting a time-based kill switch in your code.

Reply:
Maybe don’t set the kill timer in an interactive session?

Reply:
AFAIK, we don't provide an equivalent of the Windows `IsDebuggerPresent` ccall for Unix.